src/main/java/appeng/util/helpers/ItemComparisonHelper.java

```diff
@@ -0,0 +1,443 @@
+		 */
+
+		// test damageable items..
+		if( a.getItem() != null && b.getItem() != null && a.getItem().isDamageable() && a.getItem() == b.getItem() )
+		{
+			try
+			{
+				if( mode == FuzzyMode.IGNORE_ALL )
+				{
+					return true;
+				}
+				else if( mode == FuzzyMode.PERCENT_99 )
+				{
+					final Item ai = a.getItem();
+					final Item bi = b.getItem();
+
+					return ( ai.getDurabilityForDisplay( a ) > 1 ) == ( bi.getDurabilityForDisplay( b ) > 1 );
+				}
+				else
+				{
+					final Item ai = a.getItem();
+					final Item bi = b.getItem();
+
+					final float percentDamagedOfA = 1.0f - (float) ai.getDurabilityForDisplay( a );
+					final float percentDamagedOfB = 1.0f - (float) bi.getDurabilityForDisplay( b );
+
+					return ( percentDamagedOfA > mode.breakPoint ) == ( percentDamagedOfB > mode.breakPoint );
+				}
+			}
+			catch( final Throwable e )
+			{
+				if( mode == FuzzyMode.IGNORE_ALL )
+				{
+					return true;
+				}
+				else if( mode == FuzzyMode.PERCENT_99 )
+				{
+					return ( a.getItemDamage() > 1 ) == ( b.getItemDamage() > 1 );
+				}
+				else
+				{
+					final float percentDamagedOfA = (float) a.getItemDamage() / (float) a.getMaxDamage();
+					final float percentDamagedOfB = (float) b.getItemDamage() / (float) b.getMaxDamage();
+
+					return ( percentDamagedOfA > mode.breakPoint ) == ( percentDamagedOfB > mode.breakPoint );
+				}
+			}
+		}
+
+		final OreReference aOR = OreHelper.INSTANCE.isOre( a );
+		final OreReference bOR = OreHelper.INSTANCE.isOre( b );
+
+		if( OreHelper.INSTANCE.sameOre( aOR, bOR ) )
+		{
+			return true;
+		}
+
+		/*
+		 * // test ore dictionary.. int OreID = getOreID( a ); if ( OreID != -1 ) return OreID == getOreID( b );
+		 * if ( Mode != FuzzyMode.IGNORE_ALL ) { if ( a.hasTagCompound() && !isShared( a.getTagCompound() ) ) { a =
+		 * Platform.getSharedItemStack( AEItemStack.create( a ) ); }
+		 * if ( b.hasTagCompound() && !isShared( b.getTagCompound() ) ) { b = Platform.getSharedItemStack(
+		 * AEItemStack.create( b ) ); }
+		 * // test regular items with damage values and what not... if ( isShared( a.getTagCompound() ) && isShared(
+		 * b.getTagCompound() ) && a.itemID == b.itemID ) { return ((AppEngSharedNBTTagCompound)
+		 * a.getTagCompound()).compareFuzzyWithRegistry( (AppEngSharedNBTTagCompound) b.getTagCompound() ); } }
+		 */
+
+		return a.isItemEqual( b );
+	}
+
+	/**
+	 * recursive test for NBT Equality, this was faster then trying to compare / generate hashes, its also more reliable
+	 * then the vanilla version which likes to fail when NBT Compound data changes order, it is pretty expensive
+	 * performance wise, so try an use shared tag compounds as long as the system remains in AE.
+	 */
+	public boolean isNbtTagEqual( final NBTBase left, final NBTBase right )
+	{
+		// same type?
+		final byte id = left.getId();
+		if( id == right.getId() )
+		{
+			switch( id )
+			{
+				case 10:
+				{
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Extract this nested code block into a method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1199)